src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesOperation.cs

```diff
@@ -10,33 +10,80 @@ namespace Raven.Client.Documents.Operations.TimeSeries
 {
     public class TimeSeriesOperation
     {
+        private bool _appendSorted;
+        private List<AppendOperation> _appends;
+
+        public class AppendOperationEqualityComparer : IEqualityComparer<AppendOperation>
+        {
+            public static AppendOperationEqualityComparer Instance = new AppendOperationEqualityComparer();
+
+            public bool Equals(AppendOperation x, AppendOperation y)
+            {
+                if (ReferenceEquals(x, y))
+                    return true;
+                if (ReferenceEquals(x, null))
+                    return false;
+                if (ReferenceEquals(y, null))
+                    return false;
+                if (x.GetType() != y.GetType())
+                    return false;
+                return x.Timestamp.Equals(y.Timestamp) && x.Tag.Equals(y.Tag, StringComparison.Ordinal);
```

Comment:
```suggestion
                return x.Timestamp.Equals(y.Timestamp) && x.Tag.Equals(y.Tag, StringComparison.OrdinalIgnoreCase);
```

I think that we should be case insensitive here. @aviv86 thoughts?

Reply:
agreed